Migrate module from /as-aws-modules/ec2-ubuntu

JonRoma opened the pull request.

ddriddle commented on data.tf:
```diff
@@ -0,0 +1,62 @@
+}
+
+data "aws_security_groups" "selected" {
+  count = "${length(var.security_groups) > 0 ? 1 : 0}"
```
I think this should be:

count = "${length(var.security_groups)}"

ddriddle commented on variables.tf:
```diff
@@ -0,0 +1,80 @@
+  default     = "true"
+}
+
+variable "aws_vpc" {
```
Change `aws_vpc` to just `vpc`.

ddriddle commented on route53.tf:
```diff
@@ -0,0 +1,18 @@
+  zone_id = "${element(data.aws_route53_zone.selected.*.zone_id, count.index)}"
+  name    = "${lookup(var.alias[count.index], "hostname", "")}"
+  type    = "A"
+  ttl     = "${var.dns_ttl}"
```
Move `dns_ttl` into the `alias` map.

ddriddle commented on variables.tf:
```diff
@@ -0,0 +1,80 @@
+  default     = []
+}
+
+variable "dns_ttl" {
```
Remove `dns_ttl`, and add `ttl` var into `alias` map.

ddriddle reviewed (changes requested):
Take a look the comments. Overall looks good. Kind of tired, and I would not mind taking another look with fresh eyes :D

ddriddle commented on README.md:
```diff
@@ -0,0 +1,100 @@
+# ec2
+
```
Add .drone.yml and badge please :D

ddriddle replied:
Please use this `.drone.yml` file:

https://raw.githubusercontent.com/techservicesillinois/terraform-aws-acm-certificate/master/.drone.yml

And here is the code for the badge:

```
[![Build Status](https://drone.techservices.illinois.edu/api/badges/techservicesillinois/terraform-aws-efs/status.svg)](https://drone.techservices.illinois.edu/techservicesillinois/terraform-aws-efs)
```

ddriddle commented on data.tf:
```diff
@@ -0,0 +1,62 @@
+locals {
+  fqdn      = "${element(concat(data.aws_efs_file_system.default.*.dns_name, list("")), 0)}"
```
change `fqdn` to `efs_fqdn` below your output is `fqdn_efs`. Be consistent with what you choose :D

ddriddle commented on README.md:
```diff
@@ -0,0 +1,100 @@
+# ec2
+
+Provides an Elastic File System volume with mount targets and security groups,
```
This needs work!

JonRoma committed:
Create module to provide a generic EC2 instance

*   Migrate existing code from /as-aws-modules/ec2-ubuntu.  Dropped
    the "ubuntu" from the module name, since we no longer hardcode an
    Ubuntu AMI from the Terraform code, though Ubuntu remains the default.

*   Added an `ami_name_filter` variable, the default value being

        `ubuntu/images/hvm-ssd/ubuntu-bionic-18.04-amd64-server-*`

*   Added an `ami_virtualization_type_filter`, whose default is `hvm`.

*   Added an `ami_image_owner` variable (default is the Ubuntu account ID;
    see variables.tf for details.

*   Added `ports` variable. This is a list, and can be to specify TCP ports
    to open to the `cidr_block` specified in the module call. By default,
    this list consists of a single port (port 22 for SSH).

*   Added `tier` variable (e.g., public, private, nat), which is used to
    derive subnets to be associated with instance.

*   Added `template_file` variable to specify setup script passed into
    `user_data`. This user data is used to provision the EC2 instance
    upon first instantiation.

*   Allow caller to specify an EFS filesystem by ID. This allows
    deriving `efs_filesystem_dns_name`, which in turn makes the EFS
    hostname available to the template, and as a module output.

    I would have preferred to specify by filesystem name, but the
    Terraform data source (or the underlying Amazon API) doesn't
    support lookup based on that attribute.

*   Add template support. For now, ${efs_filesystem_name} is the
    only value passed to user_data when rendering the template.

*   Restructured directory since we've added more data sources and
    outputs.

*   Create an "A" record in Route53 to allow the hostname to persist
    across destroying the EC2 instance.

*   Add documentation and change default behavior and variable names.
